### ISSUE TITLE
Copy to clipboard not working

### ISSUE THREAD

**Issue by jdkcn** [linuxdeepin/deepin-screen-recorder#39]
If i check the copy to clipboard option. screenshot very slow, and not copy.

➜  ~ deepin-screen-recorder 
2020-05-20, 17:13:42.354 [Warning] [dapplication.cpp     Dtk::Widget::DApplicationPrivate::loadTranslator 295] "dtkwidget" can not find qm files
2020-05-20, 17:13:42.355 [Debug  ] [                                                         0] ConfigSettings: "effect" "is_blur" QVariant(bool, false)
2020-05-20, 17:13:42.357 [Debug  ] [                                                         0] ConfigSettings: "effect" "is_mosaic" QVariant(bool, false)
2020-05-20, 17:13:42.357 [Debug  ] [                                                         0] Setting file: "/home/rory/.config/deepin/deepin-screen-recorder.conf"
2020-05-20, 17:13:42.358 [Debug  ] [                                                         0] config changed
2020-05-20, 17:13:42.358 [Debug  ] [                                                         0] ConfigSettings: "common" "themeType" QVariant(int, 2)
2020-05-20, 17:13:42.386 [Debug  ] [                                                         0] is zhao xin: ""
2020-05-20, 17:13:42.528 [Debug  ] [                                                         0] screen rect: QRect(0,0 2160x1440)
No appenders assotiated with category qt.qpa.xcb
[Warning] <> QXcbConnection: XCB error: 5 (BadAtom), sequence: 497, resource id: 0, major code: 20 (GetProperty), minor code: 0
2020-05-20, 17:13:46.049 [Debug  ] [                                                         0] Cards changed: ""
2020-05-20, 17:13:46.105 [Debug  ] [                                                         0] mic status false
2020-05-20, 17:13:50.160 [Debug  ] [                                                         0] release event !!!
2020-05-20, 17:13:50.221 [Debug  ] [                                                         0] shotCurrentImg shotFullScreen
2020-05-20, 17:13:50.221 [Debug  ] [                                                         0] m_backgroundRect QRect(0,0 1440x960)
2020-05-20, 17:13:50.261 [Debug  ] [                                                         0] m_resultPixmap QRect(0,0 2160x1440)
2020-05-20, 17:13:50.261 [Debug  ] [                                                         0] 588 , 342 , 505 , 295 QRect(0,0 2160x1440) 1.5
2020-05-20, 17:13:50.262 [Debug  ] [                                                         0] release event !!!
2020-05-20, 17:13:50.271 [Debug  ] [                                                         0] ConfigSettings: "common" "default_savepath" QVariant(QString, "/home/rory/Pictures")
2020-05-20, 17:13:50.272 [Debug  ] [                                                         0] "/home/rory/Pictures/Screen Capture_select-area_20200520171350.png"
2020-05-20, **17:13:50.295** [Debug  ] [                                                         0] clip board success!
2020-05-20, **17:14:15.125** [Debug  ] [                                                         0] saveFilePath: "/home/rory/Pictures/Screen Capture_select-area_20200520171350.png"
2020-05-20, 17:14:25.436 [Debug  ] [                                                         0] release event !!!

**Comment by jdkcn** [comment]
Manjaro KDE Plasma 

Linux office-laptop 5.4.40-1-MANJARO #1 SMP PREEMPT Sun May 10 14:17:40 UTC 2020 x86_64 GNU/Linux

deepin-screen-recorder 5.8.0.11

**Comment by RaitoBezarius** [comment]
Can reproduce, I tried to check `xclip -t TARGETS -selection <anything> -o` and found no images, though I can confirm that Chromium can store image/png in the clipboard selection.
Confirmed with the debug that we have "clipboard success!".

`deepin-screen-recorder 0509`
`Linux 5.6.13-arch1-1 #1 SMP PREEMPT Thu, 14 May 2020 06:52:53 +0000 x86_64 GNU/Linux`

**Comment by RaitoBezarius** [comment]
Okay, investigated the source code, so, when it flush the image to the clipboard, the Qt app exits immediately, thus discarding the contents of the clipboard if you don't have any clipboard manager (theoretically, Klipper, etc.).

So I thought using a clipboard manager would just fix it, but I was absolutely unable to make it work using ClipMan, ClipIt, Greenclip.

So, I don't know if it's due to Qt or something which is done bad in the code (I tried to force event to be sent, processEvents, etc. Thought that the event loop was not properly executed).

So as an experiment, I tried `watch -n0.1 -p "xclip -o -selection clipboard -t TARGETS"` to see what was going on. Indeed, when the screenshot is taken, we can see that the proper TARGETS appear, I even succeed to see the image/png output. But that does not stay in the clipboard as of how X11 works.

So, what's the fix? I'm no X11 expert and really just want a way to make this work, clipboard manager would be optimal, but AFAIK, just piping the contents in xclip is *enough*.
So basically, I reused this patch: https://aur.archlinux.org/cgit/aur.git/tree/deepin-screenshot-copy.patch?h=deepin-screenshot-copy-patch and made it a bit more reliable, and it works fine.

I wonder how upstream wants to manage this or fix this, but here's a workaround, I'll try to make an AUR package soon on this. Ideally, it'd be good to also have a flag `-nosave` to just have clipboard-only screens.

**Comment by jdkcn** [comment]
@RaitoBezarius Thanks, Sorry for late reply. I use klipper for clipboard manage with no lucky.

**Comment by RaitoBezarius** [comment]
Indeed, the simplest thing is to apply a similar patch to the one I mentioned above and ensure that xclip is available, then it'll work. I made it work using this kind of patching.
I can add a package in Arch Linux's AUR and call it a day, but that'll fix it only for AUR users.

**Comment by jdkcn** [comment]
Ok, I use the AUR https://aur.archlinux.org/packages/deepin-screenshot-copy-patch now. 
Wait for the next working version of deepin-screen-recorder.

**Comment by EugenMayer** [comment]
https://aur.archlinux.org/packages/deepin-screenshot-copy-patch is broken right now due to the QT dependencies being newer

**Comment by bnku** [comment]
What about this issue? Is there any working solution?

**Comment by mateuszkozakiewicz** [comment]
This is still an issue, patched version from [the AUR](https://aur.archlinux.org/packages/deepin-screenshot-copy-patch) is not working for me and has visual artifacts (missing icons). For now it seems it is best to use deprecated deepin-screenshot.
